src/common.cpp

```diff
@@ -1873,6 +1884,63 @@ std::string get_executable_path(const char *argv0) {
     } else {
         return std::string(buff);
     }
+#elif defined(__OpenBSD__)
+    static std::string result;
+    if (!result.empty()) return result;
+    int cntp = 0;
+    struct stat st;
+    bool ok = false;
+    std::string path;
+    kinfo_file *kif = nullptr;
+    static kvm_t *kd = nullptr;
```

Comment:
Why is this static?

Reply:
Because the documentation says:

https://man.openbsd.org/kvm_open.3

"The kvm_open() and kvm_openfiles() functions both return a descriptor to be used in all subsequent kvm library calls. The library is fully re-entrant."

Normally this would be a global variable, but a static local should be faster and prefered in this case because it is the only place we are using the kvm_t * return value.

Reply:
We use this *once*, in this function, because we already have the static guard variable above.

Making this static as well just wastes memory.

Reply:
I.e. you already *have* the variable:

```c++
kd = kvm_openfiles(nullptr, nullptr, nullptr, KVM_NO_FILES, errbuf); 
```

This sets a local variable. And you then simply pass it to the next function:

```c++
if ((kif = kvm_getfiles(kd, KERN_FILE_BYPID, getpid(), sizeof(struct kinfo_file), &cntp)))
```

And because that code sets our static "result", it is run *once*.

Making kd `static` just means that it's kept around for the entire runtime of `fish`.

Reply:
All I know is everywhere in OpenBSD's source code that uses a kvm_t * variable they have it global or static. I'm sure there's a reason for this. I can do it if you really want me to, but this seems incorrect.

Reply:
Perhaps you are right about this, because I can't find anywhere they are doing it once like I am here. I'll take your word for it.

Reply:
I understand now, my bad.